test/performance/perf-upsert/src/config.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright Materialize, Inc. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use structopt::StructOpt;
+
+pub static KAFKA_SOURCE_NAME: &str = "perf_upsert_source";
+
+#[derive(Clone, Debug, StructOpt)]
+pub struct Args {
+    /// The materialized host
+    #[structopt(long, default_value = "localhost")]
+    pub materialized_host: String,
+
+    #[structopt(long, default_value = "6875")]
+    pub materialized_port: u16,
+
+    /// The total number of messages to create
+    #[structopt(long, default_value = "100000000")]
+    pub message_count: usize,
+
+    /// Number of messages to send per second
+    #[structopt(long, default_value = "8000")]
+    pub messages_per_second: usize,
+
+    /// The kafka host
+    #[structopt(long, default_value = "localhost")]
+    pub kafka_host: String,
+
+    /// The kafka port
+    #[structopt(long, default_value = "9092")]
+    pub kafka_port: u16,
+
+    #[structopt(long, default_value = "perf-upsert")]
+    pub kafka_topic: String,
+
+    /// Whether or not to delete the sources and views before starting
+    #[structopt(long)]
+    pub preserve_source: bool,
+
+    /// Whether or not the perf test should create a new source topic.
+    #[structopt(long)]
```

Comment:
I spent like an hour trying to get a decent way to do this with structopt subcommands. nothing seemed worth the duplication

Reply:
What would your ideal cli interface have looked like? This currently would look like `perf-upsert --create-topic`, right?

Reply:
yeah I guess I wanted something that if you included --create-topic would force you to include --partitions and --replication_factor, and otherwise would disallow you from using those options.

i ended up hacking that in, but I hoped that structopt subcommand functionality could do the work for me

Reply:
hard to know:

```console
$ cargo run -q --bin perf-upsert -- --create-topic     
thread 'main' panicked at 'Fatal internal error. Please consider filing a bug report at https://github.com/clap-rs/clap/issues', /Users/bwm/.cargo/registry/src/github.com-1ecc6299db9ec823/clap-2.33.0/src/app/usage.rs:457:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Reply:
ah, got it, pushed a commit that causes this to happen:

```console
$ cargo run -q --bin perf-upsert -- --create-topic          
error: The following required arguments were not provided:
    --replication-factor <replication-factor>
```

Reply:
thank you!